src/scib_metrics/utils/_silhouette.py

```diff
@@ -108,6 +139,32 @@ def silhouette_samples(X: np.ndarray, labels: np.ndarray) -> np.ndarray:
     """
     if X.shape[0] != labels.shape[0]:
         raise ValueError("X and labels should have the same number of samples")
-    intra_dist = _intra_cluster_distances(X, labels)
-    inter_dist = _nearest_cluster_distances(X, labels, np.unique(labels))
-    return jax.device_get((inter_dist - intra_dist) / jnp.maximum(intra_dist, inter_dist))
+    X, cumulative_mask, remapped_inds = _format_data(X, labels)
+    n_labels = X.shape[0]
+
+    # Compute intra-cluster distances
+    intra_dist_shuffled = _intra_cluster_distances(X).ravel()[cumulative_mask]
+    # Now unshuffle it
+    intra_dist = jnp.zeros_like(intra_dist_shuffled)
+    intra_dist = intra_dist.at[remapped_inds].set(intra_dist_shuffled)
+
+    # Compute nearest-cluster distances
+    inter_dist_per_label = jnp.inf * jnp.ones((n_labels, X.shape[1]))
+    inter_inds = jnp.triu_indices(n_labels, k=1)
+    inter_dist_a_b, inter_dist_b_a = _nearest_cluster_distances(X, inter_inds)
```

Comment:
you can unpack a single jax array like this?

Reply:
the block function is returning a tuple

Reply:
oh, then the return typing is wrong

Reply:
fixed